src/components/GeneralInfo/GeneralInformation/GeneralInformation.js

```diff
@@ -23,219 +23,210 @@ import useInsightsNavigate from '@redhat-cloud-services/frontend-components-util
+  BiosCardWrapper = BiosCard,
+  BootcImageCardWrapper = BootcImageCard,
+  InfrastructureCardWrapper = InfrastructureCard,
+  ConfigurationCardWrapper = ConfigurationCard,
+  SystemStatusCardWrapper = SystemStatusCard,
+  DataCollectorsCardWrapper = DataCollectorsCard,
+  CollectionCardWrapper = false,
+  SubscriptionCardWrapper = SubscriptionCard,
+  children,
+  navigate,
+  entity,
+  inventoryId,
+  loadSystemDetail,
+  systemProfilePrefetched = false,
+  showImageDetails = false,
+  isBootcHost = false,
+  showRuntimesProcesses = false,
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalTitle, setModalTitle] = useState('');
+  const [modalVariant, setModalVariant] = useState('small');
+  const [modalData, setModalData] = useState({
+    cells: [],
+    rows: [],
+    expandable: false,
+    filters: [],
+  });
 
-  onSort = (_event, index, direction, customRows) => {
-    const { rows } = this.state;
-    const sorted = (customRows || rows).sort((a, b) => {
-      const firstRow = a.cells || a;
-      const secondRow = b.cells || b;
-      const aSortBy = (
-        '' + (firstRow[index].sortValue || firstRow[index])
-      ).toLocaleLowerCase();
-      const bSortBy = (
-        '' + (secondRow[index].sortValue || secondRow[index])
-      ).toLocaleLowerCase();
-      return aSortBy < bSortBy ? -1 : 1;
-    });
-    this.setState({
-      rows: direction === SortByDirection.asc ? sorted : sorted.reverse(),
-    });
-  };
-
-  handleModalToggle = (
-    modalTitle = '',
-    { cells, rows, expandable, filters } = {},
-    modalVariant = 'small'
-  ) => {
-    if (rows)
-      this.onSort(undefined, expandable ? 1 : 0, SortByDirection.asc, rows);
-    if (this.state.isModalOpen) {
-      this.props.navigate(-1);
+  //Avoids duplicate profile fetch if consumer app already fetched, while staying backwards compatible
+  useEffect(() => {
+    if (!systemProfilePrefetched) {
+      loadSystemDetail?.(inventoryId || entity.id);
     }
+  }, [entity.id, inventoryId, loadSystemDetail, systemProfilePrefetched]);
 
-    this.setState(({ isModalOpen }) => ({
-      isModalOpen: !isModalOpen,
-      modalTitle,
-      cells,
-      expandable,
-      filters,
-      modalVariant,
-    }));
-  };
+  const onSort = useCallback(
+    (_event, index, direction, customRows) => {
+      const sorted = (customRows || modalData.rows).sort((a, b) => {
+        const firstRow = a.cells || a;
+        const secondRow = b.cells || b;
+        const aSortBy = (
+          '' + (firstRow[index].sortValue || firstRow[index])
+        ).toLocaleLowerCase();
+        const bSortBy = (
+          '' + (secondRow[index].sortValue || secondRow[index])
+        ).toLocaleLowerCase();
+        return aSortBy < bSortBy ? -1 : 1;
+      });
+      setModalData((prevData) => ({
+        ...prevData,
+        rows: direction === SortByDirection.asc ? sorted : sorted.reverse(),
+      }));
+    },
+    [modalData.rows]
```

Comment:
nit-pick: onSort will be re-defined always since modalData.rows is an array, not a primitive, https://react.dev/reference/react/useCallback#parameters, `Object.is([], [])` always results in false.

As an alternative, you could use `useDeepCompareCallback` like we did in compliance: https://github.com/RedHatInsights/compliance-frontend/blob/8c237e4a38955db1d6dad407276eee77e230a354/src/Utilities/hooks/useQuery/useQuery.js#L51 or just leave the function without useCallback since it's directly reliant on the state variable modalData. This is not something critical, but good to consider in a long-term perspective and in terms of code quality.